### ISSUE TITLE
questions about  body-part attention score

### ISSUE THREAD

**Issue by xz-123-new** [enlighten0707/Body-Part-Map-for-Interactiveness#2]
Thanks for your amazing work!       
I'm curious about the code here :
https://github.com/enlighten0707/Body-Part-Map-for-Interactiveness/blob/main/models/transformer.py
the code   
score_part = self.classifier_6v(out_binary[-1].transpose(0, 1)) # (bz, num_query, dim) -> (bz, num_query, 12)
score_part = score_part.view(*score_part.shape[:2], 6, 2) # (bz, num_query, 6, 2)
score_part_sel = F.softmax(score_part, -1)[..., 1] # (bz, num_query, 6)

is to calculate the attention score of all six body parts and choose some of them for further work, and i wonder if there any kinds of supervision for these scores?i.e. how could you choose the appropriate parts by this classifier? It seems that i don't see any GT labels telling us which part actually involves interaction or losses concerning it.Thanks for your work again and looking forward to your explanation!

**Comment by enlighten0707** [comment]
Our model can be trained either without or with body-part interactiveness supervision. In the current version, only the former is supported, and soon after we will add the latter training mode with body-part supervision.

In our main experiment, the model is trained **without** body-part interactiveness supervision. Considering that sample-wise body-part annotations are costly, we choose to supervise body-part simply from instance-level supervision, which is more economic and practical. 
We find that our method still learns body-part interactiveness well even when trained without body-part supervision (Sec. 5.3 ``Body-Part Interactiveness Detection'' in our paper). This is because instance-level interactiveness knowledge provides a loose constraint and guides body-parts interactiveness learning. Furthermore, sample-wise body-part interactiveness annotations may have some noise, while simply applying instance-level supervision is more flexible and immune to the annotation noise.

Besides, we also conduct an experiment **with** body-part supervision provided with HAKE. Results are shown in our paper, and related codes will be added, as mentioned above.

Thank you!

**Comment by kaijieshi7** [comment]
Good, I'm waiting for it.
